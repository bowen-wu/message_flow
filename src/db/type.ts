export interface User {
    id: string;
    avatar: string;
    alias: string;
    isOnline: boolean;
    ipAddress: string;
    port: number;
}

export enum MessageType {
    TEXT = 'text',
    IMAGE = 'image',
    SYSTEM = 'system',
}

export interface Message {
    id: string,
    user?: User,
    time: number;
    type: MessageType;
    content: string;
}
