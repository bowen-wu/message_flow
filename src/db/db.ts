import FriendAvatar from '../assets/images/friend_avatar.jpg';
import SelfAvatar from '../assets/images/self_avatar.jpg';
import { Message, MessageType, User } from './type';

const friend: User = {
  id: '2',
  avatar: FriendAvatar,
  alias: '小明',
  isOnline: true,
  ipAddress: '',
  port: 8080
};

const self: User = {
  id: '0',
  avatar: SelfAvatar,
  alias: '张三',
  isOnline: true,
  ipAddress: '',
  port: 8080
};

export const chatHistory: Message[] = [
  {
    id: '0',
    time: 1634368942485,
    type: MessageType.SYSTEM,
    content: 'You have added 小明 as your WeChat contact. Start chatting!'
  },
  {
    id: '1',
    user: friend,
    time: 1634368952485,
    type: MessageType.TEXT,
    content: 'Hi!我是张三'
  },
  {
    id: '2',
    user: self,
    time: 1634369080975,
    type: MessageType.TEXT,
    content:
      '写一个类似微信渲染消息流的组件。支持不同类型：文本/图片/系统消息，如加某个人到群组里，撤销消息等。'
  },
  {
    id: '3',
    time: 1634369395757,
    type: MessageType.SYSTEM,
    content: 'You recalled a message'
  },
  {
    id: '4',
    user: friend,
    time: 1634370139889,
    type: MessageType.IMAGE,
    content:
      'https://www.ringcentral.com/content/dam/rc-www/en_us/images/content/why-rc-ent/whyrc_ent_image_4.png'
  },
  {
    id: '5',
    user: self,
    time: 1634370341970,
    type: MessageType.IMAGE,
    content:
      'https://www.ringcentral.com/content/dam/rc-www/cn/images/content/advantages/blade_3.png'
  }
];
