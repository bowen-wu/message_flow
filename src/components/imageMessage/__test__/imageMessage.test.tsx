import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { MessageType } from '../../../db/type';
import { friend } from '../../../db/db';
import ImageMessage from '..';

describe('Test imageMessage Component', () => {
  const setup = () => {
    const time = 1634369395757;
    const lastMessageTime = time - 4 * 60 * 1000;
    const message = {
      id: '4',
      user: friend,
      time: 1634369400000,
      type: MessageType.IMAGE,
      content:
        'https://www.ringcentral.com/content/dam/rc-www/en_us/images/content/why-rc-ent/whyrc_ent_image_4.png'
    };
    return render(<ImageMessage message={message} lastMessageTime={lastMessageTime} />);
  };

  it('match snapshot', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });

  it('image on click', () => {
    setup();
    fireEvent.click(screen.getByTestId('imageMessage-img'));
    expect(screen.getByTestId('imageMessage-model-image')).toBeInTheDocument();
  });
});
