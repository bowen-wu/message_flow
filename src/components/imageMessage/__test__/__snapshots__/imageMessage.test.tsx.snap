// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test imageMessage Component match snapshot 1`] = `
<DocumentFragment>
  <div
    class="image-message"
  >
    <div
      class="user-message "
    >
      <div
        class="show-time"
        data-testid="time"
      >
        2021-10-16 15:30
      </div>
      <div
        class="user-message-main"
      >
        <div
          class="avatar"
        >
          <img
            alt=""
            src="friend_avatar.jpg"
          />
        </div>
        <div
          class="image-message-image"
          data-testid="imageMessage-img"
        >
          <img
            alt=""
            src="https://www.ringcentral.com/content/dam/rc-www/en_us/images/content/why-rc-ent/whyrc_ent_image_4.png"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
